### DISCUSSION TITLE
Configuring the Go install command with the Go manager?

### DISCUSSION THREAD

**Discussion by poperigby** [mason-org/mason.nvim#231] [Q&A]
I'm trying to create a package for [qml-lsp](https://invent.kde.org/sdk/qml-lsp). I would just use `mason-core.managers.go.packages` and pass in the Git repository, but just running `go install` from the repository root won't install `qml-lsp`, because the `qml-lsp` Go files are in `cmd/qml-lsp`. I need to run something like this instead:
```
go install --ldflags '-linkmode external -extldflags "-static"' ./cmd/qml-lsp/
```

Would it be possible to do this while still using the Go manager?

**Comment by williamboman** [comment]
Hey! What's they recommended method of installation? I couldn't find anything myself after taking a quick look 😕.

The Go manager is built around the usual `go install` use case (it does take a few extra options tied to `go install`, but none of the ones you listed). I'd probably just spawn go directly in a custom install function, like so:

```lua
---@async
---@param ctx InstallContext
install = function (ctx)
    ctx.go { "install", "--ldflags", [['-linkmode external -extldflags "-static"]], "./cmd/qml-lsp" }
end)
```

With that said, I'd really prefer if there was a more streamlined (and more importantly recommended by authors) way to install it.

**Reply by poperigby** [reply]
Yeah, it's a strange program. There's pretty much no documentation.
